examples/tv-casting-app/tv-casting-common/core/CastingPlayerDiscovery.cpp

```diff
@@ -0,0 +1,132 @@
+            DiscoveryFilter(DiscoveryFilterType::kDeviceType, deviceTypeFilter)));
+    }
+    else
+    {
+        ReturnErrorOnFailure(mCommissionableNodeController.DiscoverCommissioners());
+    }
+
+    mState = DISCOVERY_RUNNING;
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR CastingPlayerDiscovery::StopDiscovery()
+{
+    VerifyOrReturnError(mState == DISCOVERY_RUNNING, CHIP_ERROR_INCORRECT_STATE);
+    ReturnErrorOnFailure(mCommissionableNodeController.StopDiscovery());
+
+    mState = DISCOVERY_READY;
+
+    return CHIP_NO_ERROR;
+}
+
+void DeviceDiscoveryDelegateImpl::OnDiscoveredDevice(const chip::Dnssd::DiscoveredNodeData & nodeData)
+{
+
+    VerifyOrReturn(mClientDelegate != nullptr,
+                   ChipLogError(NotSpecified, "CastingPlayerDeviceDiscoveryDelegate, mClientDelegate is a nullptr"));
+
+    // convert nodeData to CastingPlayer
+    Attributes attributes;
+    strcpy(attributes.id, nodeData.resolutionData.hostName);
+
+    char port[kPortMaxLength] = {};
+    snprintf(port, sizeof(port), "%u", nodeData.resolutionData.port);
+    strcat(attributes.id, port);
+
+    strcpy(attributes.deviceName, nodeData.commissionData.deviceName);
+    strcpy(attributes.hostName, nodeData.resolutionData.hostName);
+    strcpy(attributes.instanceName, nodeData.commissionData.instanceName);
+    attributes.numIPs = (unsigned int) nodeData.resolutionData.numIPs;
+    for (unsigned j = 0; j < attributes.numIPs; j++)
+    {
+        attributes.ipAddresses[j] = nodeData.resolutionData.ipAddress[j];
+    }
+    attributes.productId  = nodeData.commissionData.productId;
+    attributes.vendorId   = nodeData.commissionData.vendorId;
+    attributes.deviceType = nodeData.commissionData.deviceType;
+
+    memory::Strong<CastingPlayer> player = std::make_shared<CastingPlayer>(attributes);
+
+    std::vector<memory::Strong<CastingPlayer>> castingPlayers = CastingPlayerDiscovery::GetInstance()->GetCastingPlayers();
+
+    // Add to or update castingPlayers
+    if (castingPlayers.size() != 0)
+    {
+        auto it =
+            std::find_if(castingPlayers.begin(), castingPlayers.end(),
+                         [&player](const memory::Strong<CastingPlayer> & castingPlayer) { return *castingPlayer == *player; });
+
+        // ID match found in castingPlayer, perfom update
+        if (it != castingPlayers.end())
+        {
+            unsigned index                                         = (unsigned int) std::distance(castingPlayers.begin(), it);
+            castingPlayers[index]                                  = player;
+            CastingPlayerDiscovery::GetInstance()->mCastingPlayers = castingPlayers;
+            ChipLogProgress(AppServer, "Updated Casting Player");
+
+            mClientDelegate->HandleOnUpdated(player);
+            return;
+        }
+    }
+
+    castingPlayers.push_back(player);
+    CastingPlayerDiscovery::GetInstance()->mCastingPlayers = castingPlayers;
+    mClientDelegate->HandleOnAdded(player);
+    return;
```

Comment:
There is CI check failure because this return is redundant: [https://github.com/project-chip/connectedhomeip/actions/runs/6191952094/job/16811156697?pr=29197](https://github.com/project-chip/connectedhomeip/actions/runs/6191952094/job/16811156697?pr=29197)

```
INFO TIDY ../../examples/tv-casting-app/tv-casting-common/core/CastingPlayerDiscovery.cpp: /Users/runner/work/connectedhomeip/connectedhomeip/out/default/../../examples/tv-casting-app/tv-casting-common/core/CastingPlayerDiscovery.cpp:127:5: error: redundant return statement at the end of a function with a void return type [readability-redundant-control-flow,-warnings-as-errors]
2823 127 | return;
2824 | ^~~~~~~
2825 128 | }
```